api/src/main/java/javax/enterprise/event/TransactionPhase.java

```diff
@@ -21,13 +21,16 @@
  * Distinguishes the various kinds of transactional {@linkplain javax.enterprise.event.Observes observer methods} from regular
  * observer methods which are notified immediately.
  * </p>
- * 
  * <p>
  * Transactional observer methods are observer methods which receive event notifications during the before or after completion
  * phase of the transaction in which the event was fired. If no transaction is in progress when the event is fired, they are
  * notified at the same time as other observers.
+ * If the transaction is in progress, but {@link javax.transaction.Synchronization} callback cannot be registered due to the transaction is already
```

Comment:
I would change the sentence a bit (`due to the transaction is` sounds weird). How about this:
`If the transaction is in progress, but {@link javax.transaction.Synchronization} callback cannot be registered due to the transaction being already marked for rollback or in state where {@link javax.transaction.Synchronization} callbacks cannot be registered, ... `